crates/biome_js_analyze/src/lint/nursery/no_bitwise_operators.rs

```diff
@@ -0,0 +1,167 @@
+use biome_analyze::{
+    Ast, Rule, RuleDiagnostic, RuleSource, context::RuleContext, declare_lint_rule,
+};
+use biome_console::markup;
+use biome_deserialize_macros::Deserializable;
+use biome_js_syntax::{
+    JsAssignmentExpression, JsAssignmentOperator, JsBinaryExpression, JsBinaryOperator,
+    JsSyntaxToken, JsUnaryExpression, JsUnaryOperator,
+};
+use biome_rowan::{AstNode, declare_node_union};
+#[cfg(feature = "schemars")]
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+declare_lint_rule! {
+    /// Disallow bitwise operators.
+    ///
+    /// The use of bitwise operators in JavaScript is very rare and often `&` or `|` is simply a mistyped `&&` or `||`,
+    /// which will lead to unexpected behavior.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// let x = y | z;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// x |= y;
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// let x = y || z;
+    /// ```
+    ///
+    /// ```js
+    /// let x = y && z;
+    /// ```
+    ///
+    /// ## Options
+    ///
+    /// The rule provides the options described below.
+    ///
+    ///
+    /// ```json,options
+    /// {
+    ///     "options": {
+    ///         "allow": ["&", "|", "^", "~", "<<", ">>", ">>>"]
+    ///     }
+    /// }
+    /// ```
+    /// ### allow
+    ///
+    /// Allows a list of bitwise operators to be used as exceptions.
```

Comment:
``````suggestion
    /// The rule provides the options described below.
    ///
    /// ### allow
    ///
    /// Allows a list of bitwise operators to be used as exceptions.
    ///
    /// ```json,options
    /// {
    ///     "options": {
    ///         "allow": ["&", "|", "^", "~", "<<", ">>", ">>>"]
    ///     }
    /// }
    /// ```    
``````

We usually follow a pattern we first describe the example, and then we show the code.